File: Data Structures & Algorithms/find-duplicate-integer/submission-1.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        // unordered_map<int, int> un;
        // for(int i=0; i<nums.size(); i++){
        //     un[nums[i]]++;

        //     if(un[nums[i]] == 2) return nums[i];
        // }
        // return 0;
        for(int i=0; i<nums.size(); i++){
            if(nums[abs(nums[i])] < 0) return abs(nums[i]);

            else{
                nums[abs(nums[i])] = 0 - nums[abs(nums[i])];
            }
        }
        return 0;
    }
};

File: Data Structures & Algorithms/kth-largest-element-in-an-array/submission-1.py
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        arr = [a for a in nums]

        import heapq
        heapq.heapify(arr)
        n = len(arr)
        for _ in range(n - k):
            heapq.heappop(arr)
        
        return arr[0] 